crates/biome_js_analyze/src/lint/style/use_consistent_type_definitions.rs

```diff
@@ -0,0 +1,247 @@
+    ///
+    /// ```json,options
+    /// {
+    ///     "options": {
+    ///         "style": "interface"
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// ### style
+    ///
+    /// The style to use:
+    /// - `interface`: Prefer using `interface` for object type definitions.
+    /// - `type`: Prefer using `type` for object type definitions.
+    ///
+    /// Default: `interface`
+    ///
+    pub UseConsistentTypeDefinitions {
+        version: "2.1.0",
+        name: "useConsistentTypeDefinitions",
+        language: "ts",
+        sources: &[RuleSource::EslintTypeScript("consistent-type-definitions").same()],
+        recommended: false,
+        severity: Severity::Information,
+        fix_kind: FixKind::Unsafe,
+    }
+}
+
+#[derive(Debug, Clone)]
+enum DeclarationType {
+    Interface(TsInterfaceDeclaration),
+    TypeAlias(TsTypeAliasDeclaration),
+}
+
+impl Rule for UseConsistentTypeDefinitions {
+    type Query = Ast<AnyJsDeclarationClause>;
+    type State = DeclarationType;
+    type Signals = Option<Self::State>;
+    type Options = UseConsistentTypeDefinitionsOptions;
+
+    fn run(ctx: &RuleContext<Self>) -> Option<Self::State> {
+        let query = ctx.query();
+        let options = ctx.options();
+
+        match query {
+            AnyJsDeclarationClause::TsInterfaceDeclaration(interface_decl) => {
+                if options.style == ConsistentTypeDefinition::Type {
+                    // Check if interface can be converted to type alias
+                    if can_convert_interface_to_type(interface_decl) {
+                        return Some(DeclarationType::Interface(interface_decl.clone()));
+                    }
+                }
+                None
+            }
+            AnyJsDeclarationClause::TsTypeAliasDeclaration(type_alias) => {
+                if options.style == ConsistentTypeDefinition::Interface {
+                    // Check if type alias can be converted to interface
+                    if can_convert_type_to_interface(type_alias) {
+                        return Some(DeclarationType::TypeAlias(type_alias.clone()));
+                    }
+                }
+                None
+            }
+            _ => None,
+        }
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        let options = ctx.options();
+
+        let (range, current_style, preferred_style) = match state {
+            DeclarationType::Interface(interface_decl) => {
+                (interface_decl.range(), "interface", "type")
+            }
+            DeclarationType::TypeAlias(type_alias) => (type_alias.range(), "type", "interface"),
+        };
+
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            range,
+            markup! {
+                "Use "<Emphasis>{preferred_style}</Emphasis>" instead of "<Emphasis>{current_style}</Emphasis>"."
+            },
```

Comment:
Diagnostics must follow the rule pillars: https://biomejs.dev/linter/#rule-pillars

In this case, we need a second message that explains the error. We should do this:
- first message "Use of the {current_style} detected."
- second message: "The codebase should use a consistent coding style for the definition of types. This improves the readability and consistency."